### PR TITLE
US132-TA288: Remove email sending from submitVisitorOrder pipeline an…

### PR DESCRIPTION
**Pull request by yuya-yuliya**
…d set up Marketing Automation Plan

- add EXM SMTP settings
- add New Order Placed templates
- add New Order Placed Layout
- add Order Confirmation controller rendering and ExmOrdersController

### CODE REVIEW

File: src/Feature/Checkout/code/Context/ExmContext.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+namespace HCA.Feature.Checkout.Context
+{
+    using Foundation.DependencyInjection;
+
+    [Service(typeof(IExmContext), Lifetime = Lifetime.Transient)]
+    public class ExmContext : IExmContext
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Context/ExmContext.cs:20]: Exclude from coverage

File: src/Foundation/Commerce/code/App_Config/Include/Foundation/HCA/Foundation.Commerce.config [hunk 2]
```diff
@@ -1,14 +1,32 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 
-<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
+<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/"
+               xmlns:role="http://www.sitecore.net/xmlconfig/role/"
+               xmlns:eds="http://www.sitecore.net/xmlconfig/eds/"
+               xmlns:exmEnabled="http://www.sitecore.net/xmlconfig/exmEnabled/">
   <sitecore>
+    <!-- 
+    Example of previous order confirmation email implementation
```
Comment (larionovadaria) [inline src/Foundation/Commerce/code/App_Config/Include/Foundation/HCA/Foundation.Commerce.config:9]: Create separate task to remove obsolete email sender

File: src/Project/HCA/code/App_Config/Include/Project/HCA/Project.HCA.Serialization.config [hunk 3]
```diff
@@ -15,6 +15,16 @@
...
+
+            <!-- Emails -->
+            <include name="Layouts.Emails" database="master" path="/sitecore/layout/Layouts/System/Email/HCA/Project/HCA" />
+            <include name="PlaceholderSettings.Emails" database="master"
+                     path="/sitecore/layout/Placeholder Settings/Email/HCA/Project/HCA" />
+            <include name="Branches.Emails" database="master" path="/sitecore/templates/Branches/System/Email/Messages/HCA/Project/HCA" />
+
+            <!-- TODO: discuss whether it should be a part of Shared project -->
+            <include name="AutomationPlans" database="master" path="/sitecore/system/Marketing Control Panel/Automation Plans" />
+            <include name="Compaigns" database="master" path="/sitecore/system/Marketing Control Panel/Campaigns" />
```
Comment (larionovadaria) [inline src/Project/HCA/code/App_Config/Include/Project/HCA/Project.HCA.Serialization.config:27]: Campaigns?

File: src/Project/HCA/serialization.content/Content/HCA/HCA Email.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ID: "4ed70d45-b821-461d-bba8-3407b7b1d48e"
+Parent: "bbf003f2-f4d3-415f-8136-357b71508316"
+Template: "cf9c8a2a-2794-4fea-980a-ef8426f3d6c3"
+Path: /sitecore/content/HCA/HCA Email
+DB: master
+BranchID: "f8fae92a-4f10-470c-ad5c-7ed9e6b636af"
+SharedFields:
+- ID: "1b963507-6176-4336-a14d-d5070c3b0286"
+  Hint: Base URL
+  Value: "http://hca.local"
```
Comment (larionovadaria) [inline src/Project/HCA/serialization.content/Content/HCA/HCA Email.yml:11]: Why host is configured in sitecore items? For new env we have update this item?

File: src/Project/HCA/serialization.content/Content/HCA/HCA Email.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  Hint: __Originator
+  Value: "{DB7A8703-C5AE-41FF-8B23-264195634B9E}"
+Languages:
+- Language: en
+  Versions:
+  - Version: 1
+    Fields:
+    - ID: "0f383cd8-c031-4a9d-9ed0-d55217f91287"
+      Hint: From Address
+      Value: hca@devmail.epam.com
```
Comment (larionovadaria) [inline src/Project/HCA/serialization.content/Content/HCA/HCA Email.yml:32]: Also here. Can we change it to "no-reply@devmail.epam.com"

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Feature/Checkout/code/Controllers/ExmOrdersController.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+{
+    using System.Web.Mvc;
+
+    using Glass.Mapper.Sc.Web.Mvc;
+
+    using Sitecore.Diagnostics;
+
+    using Utils;
+
+    public class ExmOrdersController : Controller
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Controllers/ExmOrdersController.cs:25]: Maybe we should rename this controller, it's quite not obvious what it actually does.

File: src/Project/HCA/serialization.content/Content/HCA/HCA Email.yml (hunk 5 above)
Comment (yuya-yuliya) [reply src/Project/HCA/serialization.content/Content/HCA/HCA Email.yml:32]: Yes, we can

**Pull request merged**